Derive inline create permissions from capabilities

oliverguenther opened the pull request:
Adds new capabilities to derive whether adding or editing work packages is allowed. Uses theses capabilities to show the add and link existing buttons in inline-create forms.

https://community.openproject.org/wp/43834

oliverguenther committed:
Derive inline create permissions from capabilities

oliverguenther committed:
Clarify interface to get contextless actions

Global != allowed_to_globally?

oliverguenther committed:
Make spec more robust

b12f commented on frontend/src/app/core/current-user/current-user.service.ts:
```diff
@@ -75,7 +75,7 @@ export class CurrentUserService {
    * Returns the set of capabilities for the given context and/or actions
    */
-  public capabilities$(actions:string[] = [], projectContext:string|null = null):Observable<ICapability[]> {
+  public capabilities$(actions:string[] = [], projectContext:string|null):Observable<ICapability[]> {
```
In this case we might as well work with `undefined` instead of `null`, I just figured it'd be better to make the "unset" option more explicit.

oliverguenther replied:
yes indeed both would work, but I kind of liked an explicit "null" more. Completely subjective though

b12f replied:
What I mean is that if you remove the defaulting to null, then we are automatically using `undefined`, so if we want to accept `null` as a value here, then I'd argue it should default to null

oliverguenther replied:
why are we automatically using `undefined` ? It's a required arg, so TS will complain if you pass undefined.

b12f replied:
Ah you're absolutely right. Bit of a brainfart I'm sorry :D

b12f reviewed (approved):
LGTM. Is the authorizationservice still relevant or can we just delete it?

oliverguenther committed:
Harden copy spec

When clicking the button too fast or trying to set ckeditor
description internally, this will fail if ckeditor is not yet initialized.

oliverguenther committed:
Add retry_block to project_autocomplete_spec.rb:95
